### PR TITLE
Tests for example configs

### PR DESCRIPTION
**Pull request by JonathanDHarris**
## Description
Aiming to resolve issue #122 and have the example configs covered by tests

Firstly, I've moved the examples down a level into the skypy directory so that they should get picked up as the unit tests run

I've added a very basic test for each example config: it really just checks the config can be used in a pipeline with no exceptions thrown and produces an output of non-zero length.

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [ ] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by JonathanDHarris** [comment]
@ntessore  are you happy to assign yourself to review this?

**Comment by ntessore** [comment]
Hi @JonathanDHarris, thanks for looking into this!

The way SkyPy is packaged, it would be important that the `examples` folder remains outside of the `skypy` module. I think what you can do is move your `skypy/examples` folder back to `examples`, and move your test cases into a single global `skypy/tests/test_examples.py` file. If you tweak the relative paths to the example configs, everything else should be good to go.

**Comment by JonathanDHarris** [comment]
No I've made the change, the test for oldest supported version is failing

```
 except (ModuleNotFoundError, AttributeError) as e:
>           raise ImportError(f'{e}\n{node.start_mark}') from e
E           ImportError: module 'astropy.modeling.models' has no attribute 'Exponential1D'
E             in "../../examples/mccl_galaxies.yml", line 8, column 15
```

I've not had time to look into it properly yet, but I'm going to guess that in that version the example config is actually incompatible with the code.

**Comment by ntessore** [comment]
Yep you are right, this is an Astropy 3 issue. So this PR is likely done from your side, thank you very very much for the contribution! This needs a little more thought than anticipated, so we will look at merging this after we ship our next release this week, when we can figure out the infrastructure going forward.

**Review by rrjbca — APPROVED** [review]
After #353 the oldest supported version of astropy is 4.0, merging `master` fixed the test failure.

**Comment by rrjbca** [comment]
Thanks for this @JonathanDHarris! If you would like to be listed as a contributor on our next release DOI (e.g. see [here](https://zenodo.org/record/4071945)) then please add your details to our [.zenodo.json](https://github.com/skypyproject/skypy/blob/master/.zenodo.json) config file as part of this pull request (using "type": "Other"). Or if you prefer not to I'm happy to merge this as it is.

**Comment by JonathanDHarris** [comment]
Thanks @rrjbca I've added myself to the json file

File: .zenodo.json [hunk 1]
```diff
@@ -127,6 +127,12 @@
             "affiliation": "University of Toronto",
             "name": "Nathaniel Starkman",
             "type": "Other"
+        },
+        {
+            "orcid": "0000-0002-3988-8591",
+            "affiliation": "None",
```
Comment (rrjbca) [inline .zenodo.json:133]:
```suggestion
```

**Review by rrjbca — APPROVED** [review]

**Pull request merged**